h2o-core/src/main/java/water/persist/PersistManager.java

```diff
@@ -409,17 +402,44 @@ public void importFiles(String path, String pattern, ArrayList<String> files, Ar
       I[Value.HDFS].importFiles(path, pattern, files, keys, fails, dels);
     }
 
-    final String[] patterns = pattern != null && !pattern.isEmpty() ? 
-            ArrayUtils.append(importFilesPatterns, pattern) : importFilesPatterns; 
-    for (String p : patterns) {
-      if (p != null && !p.isEmpty())
-        filterFiles(p, path, files, keys, fails);
+    if (pattern != null && !pattern.isEmpty())
+      filterFiles((prefix, elements) -> matchPattern(prefix, elements, pattern), path, files, keys, fails);
+    filterMetadataFiles(path, files, keys, fails);
+  }
+
+  private void filterMetadataFiles(String path, ArrayList<String> files, ArrayList<String> keys, ArrayList<String> fails) {
+    filterFiles(new DeltaLakeMetadataFilter(), path, files, keys, fails);
+  }
+
+  static class DeltaLakeMetadataFilter implements Function2<String, ArrayList<String>, ArrayList<String>> {
+    @Override
+    public ArrayList<String> apply(String prefix, ArrayList<String> ids) {
+      if (isDeltaLogPath(prefix))
+        return ids; // users intention is to import these files
+      ArrayList<String> filteredIds = new ArrayList<>(ids.size());
+      for (String id : ids) {
+        try {
+          URI uri = URI.create(id);
+          String path = uri.getPath();
+          if (isDeltaLogPath(path))
+            continue;
+        } catch (Exception e) {
+          Log.trace("Cannot create uri", e);
```

Comment:
Shouldn't the logging level be warning. I assume that trace is off by default.... Or do we want to hide such a messages.

Reply:
good point, will improve